tests/awakener.rs

```diff
@@ -0,0 +1,89 @@
+use std::thread;
+use std::time::Duration;
+
+use mio_st::event::{Event, EventedId, Ready};
+use mio_st::poll::Awakener;
+
+mod util;
+
+use self::util::{expect_events, init_with_poller};
+
+#[test]
+fn awakener() {
+    let (mut poller, mut events) = init_with_poller();
+
+    let event_id = EventedId(10);
+    let awakener = Awakener::new(&mut poller, event_id)
+        .expect("unable to create awakener");
+
+    let handle = thread::spawn(move || {
+        awakener.wake().expect("unable to wake");
+    });
+
```

Comment:
As @eduardosm pointed out. I think the problem here is that the Awakener in the main thread (where epoll resides) is "silently" dropped.

Imagine the following scenario where time occurs linearly top-to-bottom:

* main thread starts
* thread `t` spawns
  *  wake -> write(eventfd, 1u64, 8)
  * thread `t` exits
* notice that the Awakener has been moved to the thread, so when the thread exits the eventfd fd is closed, hence silently removed from the epoll set.
* expect_events won't get any events because there is nothing there

This can be reproduced by simply (This should give the thread enough time, at least under normal load):

```suggestion
thread::sleep(Duration::from_secs(2));
```

Cloning the Awakener, hence keeping the eventfd fd alive, fixes the problem.